modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/set_permissions_command.rb

```diff
@@ -90,16 +92,16 @@ def call(path:, permissions:)
       when Net::HTTPSuccess
         doc = Nokogiri::XML(response.body)
         if doc.xpath("/d:multistatus/d:response/d:propstat[d:status[text() = 'HTTP/1.1 200 OK']]/d:prop/nc:acl-list").present?
-          ServiceResult.success
+          ServiceResult.success(result: :success)
         else
           Util.error(:error, "nc:acl properly has not been set for #{path}")
         end
       when Net::HTTPNotFound
-        Util.error(:not_found)
+        Util.error(:not_found, 'Outbound request destination not found!', response)
       when Net::HTTPUnauthorized
-        Util.error(:unauthorized)
+        Util.error(:unauthorized, 'Outbound request not authorized!', response)
       else
-        Util.error(:error)
+        Util.error(:error, 'Outbound request failed!', response)
```

Comment:
Shouldn't we use `StorageError` here, along with error source and stuff like in the one drive queries?

Reply:
I think so, I didn't want to change this at this time but we need to standardize the classes, errors, messages etc.

Reply:
Yeah, I leave it to your judgement, if it would blow the scope now and if you feel well armed to tackle that foe right now. :P